### ISSUE TITLE
[SuperEditor] - Audit example app using Flutter 3.3.x and fix any issues found

### ISSUE THREAD

**Issue by matthew-carroll** [superlistapp/super_editor#801]
Flutter 3.3 introduced breaking changes to the IME API, and possibly other areas, too.

Update `SuperEditor` and `SuperTextField` to work with Flutter 3.3.x.

EDIT:
After upgrading to 3.3.3, I'm not seeing any compile-time breakages, which is odd, because there should have been breaking IME changes.

That said, here are some issues that I am seeing:

**BLOCKED BY FLUTTER:** When moving the cursor over `SuperEditor`, we initially see a text I-beam. However, once we place the caret, the cursor turns into a default cursor and never changes back to an I-beam. This is probably related to https://github.com/flutter/flutter/issues/111425 

**BLOCKED BY FLUTTER:** When scrolling down a document with a magic mouse on Mac, we no longer have any scroll momentum. I filed https://github.com/flutter/flutter/issues/112880

**PUNT:** Task components in the example editor no longer respond to pressing `enter`. However, the other components seem to handle `enter` as usual. After further investigation, I think this problem always existed. I filed https://github.com/superlistapp/super_editor/issues/804

**WILL FIX:** In the example editor, once the popover toolbar appears, it doesn't move when the user changes the selection, or collapse the selection to a caret.

**PUNT:** The example with slivers has a somewhat janky momentum. Also, while scrolling with momentum, tapping on the content doesn't stop it from scrolling. We're stuck waiting for the scroll to stop itself. I'm going to assume this is tied to other scrolling issues and I'm going to wait for the next Flutter version to worry about this.

**WILL FIX:** The "switch focus" demo is failing. It shows a caret in the document, and in the text field.

**PUNT:** When scrolling through the drawer menu in the example app, as the scroll moves, the cursor doesn't automatically change to a pointer when scrolling over buttons. Instead, we have to stop scrolling, and then move the cursor somewhere. (I filed this: https://github.com/flutter/flutter/issues/112760). Flutter says this is fixed on master, which means it's out of our control and we won't get it until 3.4.

**WILL FIX:** The mobile demos on Mac have too much horizontal padding, which causes an ugly content flow.

**Comment by domesticmouse** [comment]
> After upgrading to 3.3.3, I'm not seeing any compile-time breakages, which is odd, because there should have been breaking IME changes.

I think you'll find the breaking changes are in Flutter 3.4.x-beta and above.

**Comment by domesticmouse** [comment]
> > I think you'll find the breaking changes are in Flutter 3.4.x-beta and above.
> 
> Do you know the tentative schedule for a 3.4 release? Is that next quarter?

We have yet to roll out our Q4 Flutter release. So, I'd guess sometime before end of year.

**Comment by matthew-carroll** [comment]
I merged in a PR addressing issues I found that weren't blocked by Flutter. I'm going to close this ticket. We have another ticket at #806 to deal with a scrolling momentum bug from Flutter 3.3.3.

**Comment by hillelcoren** [comment]
Where is the 3.4.x-beta ImeAttributedTextEditingController error being tracked? #792 is closed but I'm still seeing this error.

**Comment by matthew-carroll** [comment]
Since that breaking change ended up in 3.4 instead of 3.3.3, I'll re-open the other ticket.
